extensions/topic_view_extension.rb

```diff
@@ -19,6 +19,26 @@ def self.prepended(base)
       end
     end
 
+    def next_page
+      return super if !topic.is_post_voting?
+      return super if @filter == TopicView::ACTIVITY_FILTER
```

Comment:
I think we can extract `!topic.is_post_voting? || @filter == TopicView::ACTIVITY_FILTER` to another method so we don't duplicate it across two methods.